### PR TITLE
General improvements

### PR DESCRIPTION
**Pull request by fiunchinho**
This PR:

- Use patch when deleting the finalizer from the `MachinePools`.
- Use right format on dates when logging.
- Don't requeue `MachinePools` that are not being deleted. When they are deleted, they will be queued anyway.
- Give RBAC permissions to watch `Secrets` to avoid `client-go` errors when managing the client cache for `Secrets`.

### Checklist

- [X] Update changelog in CHANGELOG.md.
- [ ] Make sure `values.yaml` and `values.schema.json` are valid.

### CODE REVIEW

File: controllers/machinepool.go [hunk 1]
```diff
@@ -10,17 +10,15 @@ import (
...
 	ctrl "sigs.k8s.io/controller-runtime"
 	ctrlclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
 	"sigs.k8s.io/controller-runtime/pkg/log"
 
 	"github.com/giantswarm/capi-garbage-collector/pkg/key"
 )
 
-var (
-	CAPIMachinePoolFinalizer = "machinepool.cluster.x-k8s.io"
```
Comment (fiunchinho) [inline controllers/machinepool.go:21]: I changed it to use the finalizer defined in the capi repository instead.

File: controllers/machinepool.go [hunk 2]
```diff
@@ -61,7 +55,7 @@ func (r *GarbageCollectorController) Reconcile(ctx context.Context, req ctrl.Req
 		return r.reconcileDelete(ctx, &machinePool, logger)
 	}
 
-	return ctrl.Result{Requeue: true, RequeueAfter: time.Minute * 5}, nil
+	return ctrl.Result{}, nil
```
Comment (fiunchinho) [inline controllers/machinepool.go:58]: This was requeing `MachinePools` that were not being deleted. We are not interested in reconciling those CRs. Should a `MachinePool` be deleted, the API will queue it for us.

File: pkg/key/key.go [hunk 3]
```diff
@@ -4,14 +4,11 @@ import (
...
-)
-
-const (
-	ClusterNameLabel = "cluster.x-k8s.io/cluster-name"
+	capi "sigs.k8s.io/cluster-api/api/v1beta1"
 )
 
 func GetClusterIDFromLabels(t v1.ObjectMeta) string {
-	return t.GetLabels()[ClusterNameLabel]
+	return t.GetLabels()[capi.ClusterLabelName]
```
Comment (fiunchinho) [inline pkg/key/key.go:11]: Using the label directly from the capi repo.

File: helm/capi-garbage-collector/templates/rbac.yaml [hunk 4]
```diff
@@ -23,6 +23,7 @@ rules:
     verbs:
       - get
       - list
+      - watch
```
Comment (fiunchinho) [inline helm/capi-garbage-collector/templates/rbac.yaml:26]: Without this the controller complains with
```
E0504 15:48:58.447071       1 reflector.go:138] pkg/mod/k8s.io/client-go@v0.24.2/tools/cache/reflector.go:167: Failed to watch *v1.Secret: unknown (get secrets)
```

**Review by bdehri — APPROVED** [review]

**Review by AndiDog — APPROVED** [review]

**Pull request merged**